lib/active_record/connection_adapters/cockroachdb/transaction_manager.rb

```diff
@@ -10,6 +10,20 @@ module TransactionManagerMonkeyPatch
         # exceeded.
         def within_new_transaction(isolation: nil, joinable: true, attempts: 0)
           super(isolation: isolation, joinable: joinable)
+        rescue ActiveRecord::ConnectionNotEstablished => e
+          raise unless retryable? error
+          raise if attempts >= @connection.max_transaction_retries
+
+          attempts += 1
+          sleep_seconds = (2 ** attempts + rand) / 10
```

Comment:
let's make a method for this `sleep_seconds` and `sleep`, it's needed by the existing error handling below, maybe something like `sleep_rand_seconds`?

Reply:
Added.